internal/controller/drplacementcontrol.go

```diff
@@ -1890,41 +1890,57 @@ func (d *DRPCInstance) newVRGSpecSync() *rmn.VRGSyncSpec {
 	}
 }
 
-func dRPolicySupportsMetro(drpolicy *rmn.DRPolicy, drclusters []rmn.DRCluster) (bool, map[string][]string) {
-	syncPeerClasses := drpolicy.Status.Sync.PeerClasses
-	aSyncPeerClasses := drpolicy.Status.Async.PeerClasses
-
-	if len(syncPeerClasses) == 0 && len(aSyncPeerClasses) == 0 {
+// dRPolicySupportsMetro returns a boolean indicating that the policy supports a Metro(Sync) DR capability, and a
+// list of list of strings, where each list of strings contains the name of the clusters that are in a Sync
+// relationship
+func dRPolicySupportsMetro(drpolicy *rmn.DRPolicy, drclusters []rmn.DRCluster, drClusterIDToName map[string]string) (
+	supportsMetro bool,
+	metroMap [][]string,
+) {
+	if len(drpolicy.Status.Sync.PeerClasses) == 0 && len(drpolicy.Status.Async.PeerClasses) == 0 {
 		return checkMetroSupportUsingRegion(drpolicy, drclusters)
 	}
 
-	if len(syncPeerClasses) != 0 {
-		return checkMetroSupportUsingPeerClass(drpolicy)
+	if len(drpolicy.Status.Sync.PeerClasses) != 0 {
+		return checkMetroSupportUsingPeerClass(drpolicy, drClusterIDToName)
 	}
 
 	return false, nil
 }
 
-func checkMetroSupportUsingPeerClass(drPolicy *rmn.DRPolicy) (bool, map[string][]string) {
-	peerClasses := drPolicy.Status.Sync.PeerClasses
-	if len(peerClasses) == 0 {
+func checkMetroSupportUsingPeerClass(drPolicy *rmn.DRPolicy, drClusterIDToName map[string]string) (bool, [][]string) {
+	if len(drPolicy.Status.Sync.PeerClasses) == 0 {
 		return false, nil
 	}
 
-	metroMap := make(map[string][]string)
-	for _, peerClass := range peerClasses {
-		metroMap[peerClass.StorageClassName] = peerClass.ClusterIDs
+	if drClusterIDToName == nil {
+		return true, nil
+	}
+
+	metroMap := [][]string{}
+
+	for idx := range drPolicy.Status.Sync.PeerClasses {
+		if len(drPolicy.Status.Sync.PeerClasses[idx].StorageID) != 1 {
+			continue
+		}
+
+		clList := []string{}
+		for _, cID := range drPolicy.Status.Sync.PeerClasses[idx].ClusterIDs {
+			clList = append(clList, drClusterIDToName[cID])
+		}
+
+		metroMap = append(metroMap, clList)
```

Comment:
We might add an empty list if `drPolicy.Status.Sync.PeerClasses[idx].ClusterIDs` is empty.
It is caught in `haveOverlappingMetroZones` but it is better to return `false, nil` if metroMap has no valid metroRegions.